### PR TITLE
fix nilptr genesisErr

### PR DESCRIPTION
**Pull request by enriavil1**
Nil ptr panic

### CODE REVIEW

File: eth/backend.go [hunk 1]
```diff
@@ -185,7 +185,11 @@ func New(stack *node.Node, config *ethconfig.Config, logger log.Logger) (*Ethere
 		}
 		var genesisErr error
 		chainConfig, genesis, genesisErr = core.WriteGenesisBlock(tx, genesisSpec, config.OverrideMergeNetsplitBlock, config.OverrideTerminalTotalDifficulty)
-		if _, ok := genesisErr.(*params.ConfigCompatError); genesisErr != nil && !ok {
+		ok := true
+		if genesisErr != nil {
+			_, ok = genesisErr.(*params.ConfigCompatError)
+		}
+		if genesisErr != nil && !ok {
```
Comment (AskAlexSharov) [inline eth/backend.go:192]: feels weird code pattern
```
if genesisErr != nil {
   if _, ok = genesisErr.(*params.ConfigCompatError); !ok {   
      return err
   }
}
```

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]
Sorry, but is meaning of this PR?
You suppress error which says: "Hey man! I see that db has another genesis! maybe you set wrong --chain parameter? maybe you set wrong --datadir parameter?" 
For me - closing eyes on this error and continue work may cause bigger problems - like writing "Goerli blocks to Mainnet database".

Or I miss-understand something?

**Comment by enriavil1** [comment]
> 



> Sorry, but is meaning of this PR? You suppress error which says: "Hey man! I see that db has another genesis! maybe you set wrong --chain parameter? maybe you set wrong --datadir parameter?" For me - closing eyes on this error and continue work may cause bigger problems - like writing "Goerli blocks to Mainnet database".
> 
> Or I miss-understand something?

I mean we are still returning the `genesisErr` saying that, but I realised that there is a case when our genesisErr causes a nil ptr reference

**Comment by AskAlexSharov** [comment]
I still don't understand meaning of this PR. 
In my head - if genesis is incompatible - better fail fast instead of hiding it.

**Pull request closed without merging**